### PR TITLE
[Merged by Bors] - chore: Golf `Finset.prod_ite_of_true`

### PR DESCRIPTION
**Pull request by YaelDillies**
... and friends. Also rename `prod_of_empty` to `prod_of_isEmpty`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 1]
```diff
@@ -1219,36 +1222,48 @@ theorem prod_ite {s : Finset α} {p : α → Prop} {hp : DecidablePred p} (f g :
 #align finset.sum_ite Finset.sum_ite
 
 @[to_additive]
-theorem prod_ite_of_false {p : α → Prop} {hp : DecidablePred p} (f g : α → β) (h : ∀ x ∈ s, ¬p x) :
-    ∏ x ∈ s, (if p x then f x else g x) = ∏ x ∈ s, g x := by
-  rw [prod_ite, filter_false_of_mem, filter_true_of_mem]
-  · simp only [prod_empty, one_mul]
-  all_goals intros; apply h; assumption
+lemma prod_dite_of_false {p : α → Prop} {_ : DecidablePred p} (h : ∀ i ∈ s, ¬ p i)
```
Comment (faenuccio) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:1225]: From the user point of view, it seems to me that the ordering `f,g,h` is more natural than `h,f,g`.

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 2]
```diff
@@ -1219,36 +1222,48 @@ theorem prod_ite {s : Finset α} {p : α → Prop} {hp : DecidablePred p} (f g :
...
 #align finset.prod_ite_of_false Finset.prod_ite_of_false
 #align finset.sum_ite_of_false Finset.sum_ite_of_false
 
 @[to_additive]
-theorem prod_ite_of_true {p : α → Prop} {hp : DecidablePred p} (f g : α → β) (h : ∀ x ∈ s, p x) :
-    ∏ x ∈ s, (if p x then f x else g x) = ∏ x ∈ s, f x := by
-  simp_rw [← ite_not (p _)]
-  apply prod_ite_of_false
-  simpa
+lemma prod_dite_of_true {p : α → Prop} {_ : DecidablePred p} (h : ∀ i ∈ s, p i) (f : ∀ i, p i → β)
```
Comment (faenuccio) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:1240]: Same comment as for the false version about the order of variables.

File: Mathlib/Analysis/Convex/Combination.lean [hunk 3]
```diff
@@ -301,8 +301,8 @@ theorem convexHull_range_eq_exists_affineCombination (v : ι → E) : convexHull
     let W : ι → R := fun i => (if i ∈ s then a * w i else 0) + if i ∈ s' then b * w' i else 0
     have hW₁ : (s ∪ s').sum W = 1 := by
       rw [sum_add_distrib, ← sum_subset subset_union_left,
-        ← sum_subset subset_union_right, sum_ite_of_true _ _ fun i hi => hi,
-        sum_ite_of_true _ _ fun i hi => hi, ← mul_sum, ← mul_sum, hw₁, hw₁', ← add_mul, hab,
+        ← sum_subset subset_union_right, sum_ite_of_true fun i hi => hi,
+        sum_ite_of_true fun i hi => hi, ← mul_sum, ← mul_sum, hw₁, hw₁', ← add_mul, hab,
         mul_one] <;> intro i _ hi' <;> simp [hi']
```
Comment (faenuccio) [inline Mathlib/Analysis/Convex/Combination.lean:306]:
```suggestion
        ← sum_subset subset_union_right, sum_ite_of_true,
        sum_ite_of_true, ← mul_sum, ← mul_sum, hw₁, hw₁', ← add_mul, hab,
        mul_one] <;> intros <;> simp_all
```

File: Mathlib/Algebra/BigOperators/Group/Finset.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/BigOperators/Group/Finset.lean:1225]: Given that `h` is an hypothesis, it should come first. Indeed, all uses in mathlib fill in `h` but don't specify `f` and `g`.

Comment (faenuccio) [reply Mathlib/Algebra/BigOperators/Group/Finset.lean:1225]: Good point.

**Comment by faenuccio** [comment]
Thanks!

**Comment by faenuccio** [comment]
maintainer merge

**Comment by robertylewis** [comment]
bors merge

**Pull request closed without merging**